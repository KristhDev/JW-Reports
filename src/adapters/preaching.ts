import { Preaching, PreachingEndpoint } from '../interfaces';

/**
 * Creates a new Preaching object from a PreachingEndpoint object.
 *
 * @param {PreachingEndpoint} preaching - The PreachingEndpoint object to convert.
 * @return {Preaching} The converted Preaching object.
 */
export const preachingAdapter = (preaching: PreachingEndpoint): Preaching => ({
    id: preaching.id,
    userId: preaching.user_id,
    day: preaching.day,
    initHour: preaching.init_hour,
    finalHour: preaching.final_hour,
    createdAt: preaching.created_at,
    updatedAt: preaching.updated_at
});