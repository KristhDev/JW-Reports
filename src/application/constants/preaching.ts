import { ParticipateInMinistry } from '@infrasturcture/interfaces';

/**
 * The headers of the table that will be displayed in the PreachingTable component
 */
export const TABLE_PREACHING_HEADERS = [
    'N°',
    'Día',
    'Hora de Inicio',
    'Hora de Fin'
];

export const HOURS_REQUIREMENTS = {
    'ninguno': 0,
    'auxiliar': 30,
    'regular': 50,
    'especial': 90
}

export const preachingMessages = {
    ADDED_SUCCESS: 'Has agregado el día de predicación correctamente.',
    DAY_REQUIRED: 'El día no puede estar vacío.',
    DELETED_SUCCESS: 'Has eliminado el día de predicación correctamente.',
    FINAL_HOUR_REQUIRED: 'La hora final no puede estar vacía.',
    INIT_HOUR_GREATER_THAN_FINAL: 'La hora inicial no puede ser mayor que la hora final.',
    INIT_HOUR_REQUIRED: 'La hora inicial no puede estar vacía.',
    MONTHLY_HOURS_REQUIRED_DONE: '¡Excelente! has cumplido con tu requerimiento de horas por mes.',
    UNSELECTED_DELETE: 'No hay un día de predicación seleccionado para eliminar.',
    UNSELECTED_UPDATE: 'No hay un día de predicación seleccionado para actualizar.',
    UPDATED_SUCCESS: 'Has actualizado el día de predicación correctamente.',
    WEEKLY_HOURS_REQUIRED_DONE: '¡Excelente! has cumplido con tu requerimiento de horas por semana.',
}

export const MINISTRY_PARTICIPATIONS: { label: string, value: ParticipateInMinistry }[] = [
    { label: 'Si', value: 'si' },
    { label: 'No', value: 'no' }
];