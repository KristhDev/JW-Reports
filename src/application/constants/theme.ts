import { ItemOption } from '../../infrasturcture/interfaces';

/**
 * The theme options to select
 */
export const THEME_OPTIONS: ItemOption[] = [
    { label: 'Modo predeterminado', value: 'default' },
    { label: 'Modo claro', value: 'light' },
    { label: 'Modo oscuro', value: 'dark' }
];