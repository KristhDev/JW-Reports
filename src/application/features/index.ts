export * from './auth';
export * from './courses';
export * from './lessons';
export * from './permissions';
export * from './preaching';
export * from './revisits';
export * from './status';
export * from './types';
export * from './ui';