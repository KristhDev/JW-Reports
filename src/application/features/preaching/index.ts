export {
    default as preachingReducer,
    INIT_PREACHING,
    PREACHING_INITIAL_STATE,
    addPreaching,
    clearPreaching,
    removePreaching,
    setIsPreachingDeleting,
    setIsPreachingLoading,
    setIsPreachingsLoading,
    setPreachings,
    setSelectedDate,
    setSelectedPreaching,
    updatePreaching
} from './slice';

export * from './types';