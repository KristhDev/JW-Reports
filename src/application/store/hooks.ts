import { useDispatch, useSelector } from 'react-redux';

/* Store */
import { AppDispatch, RootState } from './store';

/* Defined types hooks */
export const useAppDispatch = useDispatch.withTypes<AppDispatch>();
export const useAppSelector = useSelector.withTypes<RootState>();