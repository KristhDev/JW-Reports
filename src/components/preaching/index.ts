export { PreachingForm } from './PreachingForm';
export { PreachingTable } from './PreachingTable';
export { RevisitCard } from './RevisitCard';