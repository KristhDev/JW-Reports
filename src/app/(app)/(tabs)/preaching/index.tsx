import { Redirect } from 'expo-router';

/* Constants */
import { precursors } from '@application/constants';

/* Auth */
import { useAuth } from '@auth';

/**
 * Redirects to either the Precursor or Publisher tab based on the user's
 * assigned role.
 *
 * @returns A redirect to either the Precursor or Publisher tab.
 */
export default function PreachingRootScreen(): JSX.Element {
    const { state: { user } } = useAuth();

    if (user.precursor !== precursors.NINGUNO) return <Redirect href="/(app)/(tabs)/preaching/precursor" />
    else return <Redirect href="/(app)/(tabs)/preaching/publisher" />
}