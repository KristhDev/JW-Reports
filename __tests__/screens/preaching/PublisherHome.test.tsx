import React from 'react';
import { MenuProvider } from 'react-native-popup-menu';
import { render, screen } from '@testing-library/react-native';

/* Screens */
import { PublisherHome } from '../../../src/screens/preaching';

/* Features */
import { INIT_LESSON, INIT_REVISIT } from '../../../src/features';

/* Hooks */
import { useAuth, useCourses, usePreaching, useRevisits, useStatus, useTheme } from '../../../src/hooks';

/* Theme */
import { darkColors } from '../../../src/theme';

/* Mocks */
import {
    completeRevisitMock,
    deleteLessonMock,
    finishOrStartLessonMock,
    lastLessonStateMock,
    lastRevisitStateMock,
    loadLastLessonMock,
    loadLastRevisitMock,
    preachingsStateMock,
    saveCourseMock,
    setErrorFormMock,
    setSelectedCourseMock,
    setSelectedLessonMock,
    setSelectedRevisitMock,
    setStatusMock,
    testUser
} from '../../mocks';

/* Mock hooks */
jest.mock('../../../src/hooks/useAuth.ts');
jest.mock('../../../src/hooks/useCourses.ts');
jest.mock('../../../src/hooks/usePreaching.ts');
jest.mock('../../../src/hooks/useRevisits.ts');
jest.mock('../../../src/hooks/useStatus.ts');
jest.mock('../../../src/hooks/useTheme.ts');

describe('Test in <PublisherHome /> screen', () => {
    (useAuth as jest.Mock).mockReturnValue({
        state: { user: testUser },
    });

    (useCourses as jest.Mock).mockReturnValue({
        state: lastLessonStateMock,
        deleteLesson: deleteLessonMock,
        finishOrStartLesson: finishOrStartLessonMock,
        loadLastLesson: loadLastLessonMock,
        saveCourse: saveCourseMock,
        setSelectedCourse: setSelectedCourseMock,
        setSelectedLesson: setSelectedLessonMock,
    });

    (usePreaching as jest.Mock).mockReturnValue({
        state: preachingsStateMock
    });

    (useRevisits as jest.Mock).mockReturnValue({
        state: lastRevisitStateMock,
        completeRevisit: completeRevisitMock,
        loadLastRevisit: loadLastRevisitMock,
        setSelectedRevisit: setSelectedRevisitMock
    });

    (useStatus as jest.Mock).mockReturnValue({
        setStatus: setStatusMock,
        setErrorForm: setErrorFormMock
    });

    (useTheme as jest.Mock).mockReturnValue({
        state: { colors: darkColors, selectedTheme: 'dark' },
        BUTTON_TRANSLUCENT_COLOR: 'rgba(255, 255, 255, 0.15)'
    });

    it('should to match snapshot', () => {
        render(
            <MenuProvider>
                <PublisherHome />
            </MenuProvider>
        );

        expect(screen.toJSON()).toMatchSnapshot();
    });

    it('should render loadings', () => {
        (useCourses as jest.Mock).mockReturnValue({
            state: {
                ...lastLessonStateMock,
                isLastLessonLoading: true
            },
            deleteLesson: deleteLessonMock,
            finishOrStartLesson: finishOrStartLessonMock,
            loadLastLesson: loadLastLessonMock,
            saveCourse: saveCourseMock,
            setSelectedCourse: setSelectedCourseMock,
            setSelectedLesson: setSelectedLessonMock,
        });

        (useRevisits as jest.Mock).mockReturnValue({
            state: {
                ...lastRevisitStateMock,
                isLastRevisitLoading: true
            },
            completeRevisit: completeRevisitMock,
            loadLastRevisit: loadLastRevisitMock,
            setSelectedRevisit: setSelectedRevisitMock
        });


        render(
            <MenuProvider>
                <PublisherHome />
            </MenuProvider>
        );

        const lastLessonLoading = screen.queryByTestId('last-lesson-loading');
        const lastRevisitLoading = screen.queryByTestId('last-revisit-loading');

        expect(lastLessonLoading).toBeTruthy();
        expect(lastRevisitLoading).toBeTruthy();
    });

    it('should render empty messages', () => {
        (useCourses as jest.Mock).mockReturnValue({
            state: {
                ...lastLessonStateMock,
                lastLesson: INIT_LESSON
            },
            deleteLesson: deleteLessonMock,
            finishOrStartLesson: finishOrStartLessonMock,
            loadLastLesson: loadLastLessonMock,
            saveCourse: saveCourseMock,
            setSelectedCourse: setSelectedCourseMock,
            setSelectedLesson: setSelectedLessonMock,
        });

        (useRevisits as jest.Mock).mockReturnValue({
            state: {
                ...lastRevisitStateMock,
                lastRevisit: INIT_REVISIT
            },
            completeRevisit: completeRevisitMock,
            loadLastRevisit: loadLastRevisitMock,
            setSelectedRevisit: setSelectedRevisitMock
        });

        render(
            <MenuProvider>
                <PublisherHome />
            </MenuProvider>
        );

        const emptyMessages = screen.queryAllByTestId('info-text-text');

        expect(emptyMessages[0]).toBeTruthy();
        expect(emptyMessages[0].props.children).toBe('No haz agregado ninguna lección para un curso biblíco.');
        expect(emptyMessages[1]).toBeTruthy();
        expect(emptyMessages[1].props.children).toBe('No haz agregado ninguna revisita.');
    });
});