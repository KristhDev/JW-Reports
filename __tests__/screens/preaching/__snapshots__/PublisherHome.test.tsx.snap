// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <PublisherHome /> screen should to match snapshot 1`] = `
<View
  onLayout={[Function]}
  style={
    {
      "flex": 1,
    }
  }
>
  <View
    style={
      [
        {
          "flex": 1,
        },
        undefined,
        undefined,
        undefined,
      ]
    }
  >
    <RCTScrollView
      contentContainerStyle={
        {
          "alignItems": "center",
          "padding": 24,
          "paddingBottom": 100,
        }
      }
      overScrollMode="never"
      refreshControl={
        <RefreshControlMock
          colors={
            [
              "#000",
            ]
          }
          onRefresh={[Function]}
          refreshing={false}
        />
      }
      style={
        {
          "flex": 1,
        }
      }
    >
      <RCTRefreshControl />
      <View>
        <View
          style={
            [
              {
                "alignItems": "flex-start",
                "marginBottom": 40,
                "width": "100%",
              },
              {
                "marginBottom": 16,
                "paddingBottom": 0,
                "paddingTop": 8,
              },
            ]
          }
        >
          <Text
            style={
              [
                {
                  "color": "#FFFFFF",
                  "fontSize": 38,
                  "fontWeight": "bold",
                },
                {
                  "fontSize": 24,
                },
              ]
            }
            testID="title-text"
          >
            ÚLTIMA LECCIÓN
          </Text>
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": false,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            [
              {
                "overflow": "hidden",
              },
              {
                "marginVertical": 8,
                "width": 734,
              },
            ]
          }
          testID="lesson-card-touchable"
        >
          <View
            style={
              {
                "backgroundColor": "#292929",
                "padding": 15,
              }
            }
          >
            <Text
              style={
                {
                  "color": "#BFBFBF",
                  "fontSize": 14,
                  "marginBottom": 15,
                }
              }
              testID="lesson-card-status-text"
            >
              Clase para el 31 de December del 2020
            </Text>
            <Text
              style={
                {
                  "color": "#FFFFFF",
                  "fontSize": 16,
                }
              }
              testID="lesson-card-description-text"
            >
              Nemo et et non nesciunt ipsum sed rerum vero. Nostrum quae officiis qui nihil dignissimos consectetur aperiam. Vel harum ex. Ipsum ratione necessitatibus eum sint vero enim et consequatur. Ab autem re...
            </Text>
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": false,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                [
                  {
                    "overflow": "hidden",
                  },
                  [
                    {
                      "alignItems": "center",
                      "backgroundColor": "transparent",
                      "borderRadius": 50,
                      "height": 48,
                      "justifyContent": "center",
                      "margin": 5,
                      "width": 48,
                    },
                    {
                      "height": 35,
                      "position": "absolute",
                      "right": -2.5,
                      "top": 5,
                      "width": 35,
                    },
                  ],
                ]
              }
              testID="fab-touchable"
            >
              <Text
                allowFontScaling={false}
                selectable={false}
                style={
                  [
                    {
                      "color": "#C0A7E1",
                      "fontSize": 21,
                    },
                    undefined,
                    {
                      "fontFamily": "Ionicons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    {},
                  ]
                }
              >
                
              </Text>
            </View>
            <View
              style={
                {
                  "position": "absolute",
                  "right": 20,
                  "top": 30,
                }
              }
            >
              <View
                collapsable={false}
              >
                <View
                  accessibilityValue={
                    {
                      "max": undefined,
                      "min": undefined,
                      "now": undefined,
                      "text": undefined,
                    }
                  }
                  accessible={true}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                >
                  <View
                    ctx={
                      {
                        "menuActions": {
                          "_getOpenedMenu": [Function],
                          "_notify": [Function],
                          "closeMenu": [Function],
                          "isMenuOpen": [Function],
                          "openMenu": [Function],
                          "toggleMenu": [Function],
                        },
                        "menuRegistry": {
                          "getAll": [Function],
                          "getMenu": [Function],
                          "setOptionsCustomStyles": [Function],
                          "subscribe": [Function],
                          "unsubscribe": [Function],
                          "updateLayoutInfo": [Function],
                        },
                      }
                    }
                    style={
                      [
                        undefined,
                        undefined,
                      ]
                    }
                  />
                </View>
              </View>
            </View>
          </View>
        </View>
        <View
          style={
            [
              {
                "alignItems": "flex-start",
                "marginBottom": 40,
                "width": "100%",
              },
              {
                "marginBottom": 16,
                "paddingBottom": 0,
                "paddingTop": 32,
              },
            ]
          }
        >
          <Text
            style={
              [
                {
                  "color": "#FFFFFF",
                  "fontSize": 38,
                  "fontWeight": "bold",
                },
                {
                  "fontSize": 24,
                },
              ]
            }
            testID="title-text"
          >
            ÚLTIMA REVISITA
          </Text>
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": false,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            [
              {
                "overflow": "hidden",
              },
              {
                "marginVertical": 8,
                "width": 734,
              },
            ]
          }
          testID="revisit-card-touchable"
        >
          <View
            style={
              {
                "backgroundColor": "#292929",
                "padding": 16,
              }
            }
          >
            <Text
              style={
                {
                  "color": "#BFBFBF",
                  "fontSize": 14,
                  "marginBottom": 15,
                }
              }
              testID="revisit-card-next-visit-text"
            >
              Visitar el 09 de December del 2021
            </Text>
            <Text
              style={
                {
                  "color": "#FFFFFF",
                  "fontSize": 18,
                  "fontWeight": "bold",
                  "marginBottom": 5,
                }
              }
              testID="revisit-card-person-name-text"
            >
              Dariana Hand
            </Text>
            <Text
              style={
                {
                  "color": "#FFFFFF",
                  "fontSize": 16,
                }
              }
              testID="revisit-card-about-text"
            >
              Ratione quae debitis pariatur tempora magnam est voluptatibus. Eveniet ipsum omnis maxime et soluta iusto. Autem aut ducimus dolore et recusandae et inventore. Enim autem in hic optio.
            </Text>
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": false,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                [
                  {
                    "overflow": "hidden",
                  },
                  [
                    {
                      "alignItems": "center",
                      "backgroundColor": "transparent",
                      "borderRadius": 50,
                      "height": 48,
                      "justifyContent": "center",
                      "margin": 5,
                      "width": 48,
                    },
                    {
                      "height": 35,
                      "position": "absolute",
                      "right": -2.5,
                      "top": 5,
                      "width": 35,
                    },
                  ],
                ]
              }
              testID="fab-touchable"
            >
              <Text
                allowFontScaling={false}
                selectable={false}
                style={
                  [
                    {
                      "color": "#C0A7E1",
                      "fontSize": 21,
                    },
                    undefined,
                    {
                      "fontFamily": "Ionicons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    {},
                  ]
                }
              >
                
              </Text>
            </View>
            <View
              style={
                {
                  "position": "absolute",
                  "right": 20,
                  "top": 30,
                }
              }
            >
              <View
                collapsable={false}
              >
                <View
                  accessibilityValue={
                    {
                      "max": undefined,
                      "min": undefined,
                      "now": undefined,
                      "text": undefined,
                    }
                  }
                  accessible={true}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                >
                  <View
                    ctx={
                      {
                        "menuActions": {
                          "_getOpenedMenu": [Function],
                          "_notify": [Function],
                          "closeMenu": [Function],
                          "isMenuOpen": [Function],
                          "openMenu": [Function],
                          "toggleMenu": [Function],
                        },
                        "menuRegistry": {
                          "getAll": [Function],
                          "getMenu": [Function],
                          "setOptionsCustomStyles": [Function],
                          "subscribe": [Function],
                          "unsubscribe": [Function],
                          "updateLayoutInfo": [Function],
                        },
                      }
                    }
                    style={
                      [
                        undefined,
                        undefined,
                      ]
                    }
                  />
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": false,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        [
          {
            "overflow": "hidden",
          },
          [
            {
              "alignItems": "center",
              "backgroundColor": "#C0A7E1",
              "borderRadius": 50,
              "height": 48,
              "justifyContent": "center",
              "margin": 5,
              "width": 48,
            },
            {
              "bottom": 5,
              "height": 60,
              "position": "absolute",
              "right": 10,
              "width": 60,
            },
          ],
        ]
      }
      testID="fab-touchable"
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          [
            {
              "color": "#121212",
              "fontSize": 40,
            },
            undefined,
            {
              "fontFamily": "Ionicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            {},
          ]
        }
      >
        
      </Text>
    </View>
    <Modal
      animationType="fade"
      hardwareAccelerated={false}
      statusBarTranslucent={true}
      transparent={true}
      visible={false}
    />
    <Modal
      animationType="fade"
      hardwareAccelerated={false}
      statusBarTranslucent={true}
      transparent={true}
      visible={false}
    />
    <Modal
      animationType="fade"
      hardwareAccelerated={false}
      statusBarTranslucent={true}
      transparent={true}
      visible={false}
    />
    <Modal
      animationType="fade"
      hardwareAccelerated={false}
      statusBarTranslucent={true}
      transparent={true}
      visible={false}
    />
    <Modal
      animationType="fade"
      hardwareAccelerated={false}
      statusBarTranslucent={true}
      transparent={true}
      visible={false}
    />
    <Modal
      animationType="fade"
      hardwareAccelerated={false}
      statusBarTranslucent={true}
      transparent={true}
      visible={false}
    />
  </View>
  <RCTSafeAreaView
    pointerEvents="box-none"
    style={
      {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <View
      collapsable={false}
      onLayout={[Function]}
      pointerEvents="box-none"
      style={
        {
          "flex": 1,
        }
      }
    />
  </RCTSafeAreaView>
</View>
`;
